sdk/ai/Azure.AI.Projects/README.md

```diff
@@ -210,6 +212,110 @@ Agent agent = agentResponse.Value;
 With a file ID association and a supported tool enabled, the agent will then be able to consume the associated
 data when running threads.
 
+#### Create Agent with Enterprise File Search
+
+We can upload file to Azure as it is shown in the example, or use the existing Azure blob storage. In the code below we demonstrate how this can be achieved. First we upload file to azure and create `VectorStoreDataSource`, which then is used to create vector store. This vector store is then given to the `FileSearchTool` constructor.
+
+```C# Snippet:CreateVectorStoreBlob
+var ds = new VectorStoreDataSource(
+    assetIdentifier: blobURI,
+    assetType: VectorStoreDataSourceAssetType.UriAsset
+);
+var vectorStoreTask = await client.CreateVectorStoreAsync(
+    fileIds: null,
+    name: "sample_vector_store",
+    storeConfiguration: new VectorStoreConfiguration(
+        dataSources: new List<VectorStoreDataSource> { ds }
+    )
+);
+var vectorStore = vectorStoreTask.Value;
+
+FileSearchToolResource fileSearchResource = new([vectorStore.Id], null);
+
+List<ToolDefinition> tools = [new FileSearchToolDefinition()];
+Response<Agent> agentResponse = await client.CreateAgentAsync(
+    model: modelName,
+    name: "my-assistant",
+    instructions: "You are helpful assistant.",
+    tools: tools,
+    toolResources: new ToolResources() { FileSearch = fileSearchResource }
+);
+```
+
+We also can attach files to the existing vector vector store. In the code snippet below, we first create an empty vector store and add file to it.
+
+```C# Snippet:BatchFileAttachment
+var ds = new VectorStoreDataSource(
+    assetIdentifier: blobURI,
+    assetType: VectorStoreDataSourceAssetType.UriAsset
+);
+var vectorStoreTask = await client.CreateVectorStoreAsync(
+    fileIds: null,
+    name: "sample_vector_store"
+);
+var vectorStore = vectorStoreTask.Value;
+
+var uploadTask = await client.CreateVectorStoreFileBatchAsync(
+    vectorStoreId: vectorStore.Id,
+    dataSources: new List<VectorStoreDataSource> { ds }
+);
+Console.WriteLine($"Created vector store file batch, vector store file batch ID: {uploadTask.Value.Id}");
+
+FileSearchToolResource fileSearchResource = new([vectorStore.Id], null);
+```
+
+#### Create Message with Code Interpreter Attachment
+
+To attach a file with the context to the message, the `MessageAttachment` class have to be used. To be able to process the attached file contents we need to provide the `List` with the single element `CodeInterpreterToolDefinition` as a `tools` parameter to both `CreateAgent` method and `MessageAttachment` class constructor.
```

Comment:
"has to be used".

Or better, more soft language will be "To attach a file with the context to the message, use the `MessageAttachment` class. "

Reply:
Fixed